src/tx_builder/builder.rs

```diff
@@ -0,0 +1,331 @@
+    ///   * `receiver_addr` receiver address string
+    ///   * `capacity` capacity in shannon
+    pub fn add_output_raw(
+        &mut self,
+        receiver_addr: &str,
+        capacity: u64,
+    ) -> Result<(), TxBuilderError> {
+        let receiver_address =
+            Address::from_str(receiver_addr).map_err(TxBuilderError::AddressFormat)?;
+        let hum_capacity = HumanCapacity(capacity);
+        let output = Self::build_output(&receiver_address, hum_capacity);
+        self.outputs.push((output, Bytes::default()));
+        Ok(())
+    }
+
+    pub fn add_output(&mut self, receiver: &Address, capacity: HumanCapacity) {
+        let output = Self::build_output(receiver, capacity);
+        self.outputs.push((output, Bytes::default()));
+    }
+    /// Add output with data from receiver address and capacity
+    /// # Arguments:
+    ///   * `receiver_addr` receiver address string
+    ///   * `capacity` capacity in shannon
+    pub fn add_output_data(&mut self, receiver: &Address, capacity: HumanCapacity, data: Bytes) {
+        let output = Self::build_output(receiver, capacity);
+        self.outputs.push((output, data));
+    }
+
+    pub fn set_fee_rate(&mut self, fee_rate: FeeRate) {
+        self.balancer.fee_rate = fee_rate;
+    }
+
+    /// Set change lock script instead of use the same script of sender
+    pub fn set_change_lock_script(&mut self, change_lock_script: Script) {
+        self.balancer.change_lock_script = Some(change_lock_script);
+    }
+
+    /// Set change lock script instead of use the same script of sender
+    pub fn set_change_addr(&mut self, change_addr: &str) -> Result<(), TxBuilderError> {
+        let change_address =
+            Address::from_str(change_addr).map_err(TxBuilderError::AddressFormat)?;
+        self.balancer.change_lock_script = Some(Script::from(&change_address));
+        Ok(())
+    }
+    /// Set the place holder witness if the default witness not fit the requirements.
+    /// If can find a value with script, new place holder witness will be set, or the value will be appended.
+    /// # Arguments:
+    /// * `script` key to search the old place holder witness
+    /// * `placeholder_witness` the new value
+    pub fn set_placeholder_witness(&mut self, script: Script, placeholder_witness: WitnessArgs) {
+        self.balancer
+            .capacity_provider
+            .set_witness(script, placeholder_witness);
+    }
+
+    pub fn set_sender_placeholder_witness(&mut self, placeholder_witness: WitnessArgs) {
+        let script = Script::from(&self.sender);
+        self.set_placeholder_witness(script, placeholder_witness);
+    }
+
+    /// clear the change lock script, so it will use the sender's according script.
+    pub fn clear_change_lock_script(&mut self) {
+        self.balancer.change_lock_script = None;
+    }
+
+    /// insert a cell cep to cell_dep_resolver
+    pub fn insert_resolv_cell_dep(&mut self, code_hash: H256, tx_hash: H256, index: u32) {
```

Comment:
suggest to use `OutPoint` struct directly and rename to `insert_cell_dep_resolver`